src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -280,6 +280,8 @@ Copyright (c) .NET Foundation. All rights reserved.
         <Private>false</Private>
```

Comment:
existing: why use batching here instead of a transform?  If you use @(_CompileFileDefinitions->'%(ResolvedPath)') it will preserve all the meta-data and you don't need to do things like
```
 +        <NuGetPackageId>%(_CompileFileDefinitions.PackageName)</NuGetPackageId>
 +        <NuGetPackageVersion>%(_CompileFileDefinitions.PackageVersion)</NuGetPackageVersion>
```